primitives/src/math/fixed.rs

```diff
@@ -243,36 +243,35 @@ impl<F: Fixed + ToString, N: TryFrom<u128>> FromFixedToDecimal<F> for N {
     fn from_fixed_to_fixed_decimal(fixed: F, decimals: u8) -> Result<N, &'static str> {
         let decimals_usize = decimals as usize;
         let s = fixed.to_string();
-        let mut parts: Vec<&str> = s.split('.').collect();
-        // If there's no fractional part, then `fixed` was an integer.
-        if parts.len() != 2 {
-            parts.push("0");
+        let mut parts = s.split('.');
+        let int_part = parts.next().ok_or("Empty string provided")?;
+        let frac_part = parts.next().unwrap_or("0");
+        // Ensure that the iterator is now exhausted.
+        if parts.next().is_some() {
+            return Err("The string contains multiple decimal points");
         }
 
-        let (int_part, frac_part) = (parts[0], parts[1]);
-        let mut increment = false;
-
+        let mut increment_int_part = false;
         let new_frac_part = if frac_part.len() < decimals_usize {
             format!("{}{}", frac_part, "0".repeat(decimals_usize.saturating_sub(frac_part.len())))
         } else {
-            // Adding rounding behavior
-            let round_digit = frac_part.chars().nth(decimals_usize);
-            match round_digit {
-                Some(d) if d >= '5' => increment = true,
-                _ => {}
+            let round_digit = frac_part.chars().nth(decimals_usize).and_then(|c| c.to_digit(10));
+            // Note that `round_digit` might be `None`, for example if `frac_part = 245` and
+            // `decimals = 3`. There's no need to round in that case.
+            if let Some(d) = round_digit {
+                if d >= 5 {
+                    increment_int_part = true;
+                }
             }
```

Comment:
```suggestion
            let round_digit = frac_part.chars().nth(decimals_usize).and_then(|c| c.to_digit(10)).ok_or("The char at decimals_usize was not found, although the frac_part length is higher than decimals_usize.")?;
            if round_digit >= 5 {
                increment_int_part = true;
            }
```

Reply:
Yeah, I had exactly this. But it turns out it's exactly _not_ what want ;) As the comment explains, `round_digit` might be `None`, in which case we can pretend that it's zero; or, equivalently, if it's `None`, then we just don't do any rounding. You'll find that a couple of tests will fail if we make the proposed change.

Reply:
Nevermind, just saw the comment above.

Reply:
Yeah, nice suggestion. The only problem here is that we also need an `else` case, so I think it's fine to leave it the way it is.